Dockerfile

```diff
@@ -1,7 +1,8 @@
 FROM python:3.9
 
 COPY . /ml-compiler-opt
-RUN pip install --no-cache-dir -r /ml-compiler-opt/requirements.txt
+WORKDIR /ml-compiler-opt
```

Comment:
I'd move the `WORKDIR` line above the `COPY` and use the `COPY . .` pattern, that way we're not duplicating the `/ml-compiler-opt` path which is a common pattern:

```
WORKDIR /ml-compiler-opt
COPY . .
```

Reply:
Fixed. Thanks for the pointer.